src/PhpDocumentFactory.php

```diff
@@ -0,0 +1,215 @@
+<?php
+declare(strict_types = 1);
+
+namespace LanguageServer;
+
+use LanguageServer\NodeVisitor\ColumnCalculator;
+use LanguageServer\NodeVisitor\DefinitionCollector;
+use LanguageServer\NodeVisitor\DocBlockParser;
+use LanguageServer\NodeVisitor\ReferencesAdder;
+use LanguageServer\NodeVisitor\ReferencesCollector;
+use LanguageServer\Protocol\ClientCapabilities;
+use LanguageServer\Protocol\Diagnostic;
+use LanguageServer\Protocol\DiagnosticSeverity;
+use phpDocumentor\Reflection\DocBlockFactory;
+use PhpParser\ErrorHandler\Collecting;
+use PhpParser\NodeTraverser;
+use PhpParser\NodeVisitor\NameResolver;
+use Sabre\Event\Promise;
+use function Sabre\Event\coroutine;
+
+class PhpDocumentFactory
+{
+    /**
+     * Holds account of all definitions, references etc.
+     * @var Project
+     */
+    private $project;
+    
+    /**
+     * Instance of the PHP parser
+     *
+     * @var Parser
+     */
+    private $parser;
+
+    /**
+     * The DocBlockFactory instance to parse docblocks
+     *
+     * @var DocBlockFactory
+     */
+    private $docBlockFactory;
+
+    /**
+     * The DefinitionResolver instance to resolve reference nodes to Definitions
+     *
+     * @var DefinitionResolver
+     */
+    private $definitionResolver;
+
+    /**
+     * Reference to the language server client interface
+     *
+     * @var LanguageClient
+     */
+    private $client;
+
+    /**
+     * The client's capabilities
+     *
+     * @var ClientCapabilities
+     */
+    private $clientCapabilities;
+
+    public function __construct(LanguageClient $client, Project $project, ClientCapabilities $clientCapabilities = null)
+    {
+        $this->client = $client;
+        $this->clientCapabilities = $clientCapabilities ? : new ClientCapabilities();
```

Comment:
The document(factory)  should not know about clients or LSP. The content retrieval should happen over an injected interface like `ContentRetriever` then